Remove broken and useless file in build

wolfogre opened the pull request:
Related to #17684, #18633.

After the [changing](https://github.com/go-gitea/gitea/pull/18633/files#diff-5d3fa357bcf82db38c5248837eedce463890b9a5b943a8b847f41fbb2bfb5267) of `codeformat.FormatGoImports`, the `gitea-format-imports.go` became uncompilable.

And According to the comment:

https://github.com/go-gitea/gitea/blob/00d8d4a0e29d547f8aaca726a306a031513e7332/build/code-batch-process.go#L26

I think it would be clearer to rename `gitea-format-imports.go` to `gitea-fmt.go`

wxiaoguang commented:
+1.

FYI, that's a historical reason of the name "code-batch-process.go".

At that time, Windows developers can not run `make fmt` because Windows can not accept long argument strings, the command `./program arg1 arg2 ... a-lot-of-args` always fails on Windows. That's why `code-batch-process.go` comes, it splits the command line arguments into groups and run separate commands with fewer arguments: `./program arg1 arg2 ... argN` and `./program argN argN+1 ... argN+N`, and so on.

After many code changes, Gitea has dropped many old tools, and I think most modern tools do not need to use `./program arg1 arg2 ... a-lot-of-args` to pass filenames and won't have the annoying Windows command argument problem.

If it's the case, maybe it's time to retire the argument splitting function of `code-batch-process.go`

wolfogre commented:
@wxiaoguang You may may have misunderstood me, I mean rename `gitea-format-imports.go` to `gitea-fmt.go`.

wxiaoguang reviewed (dismissed):
Hmm, I saw `code-batch-process.go` in the PR comment so added more information about it (at the beginning, they are related ...), and maybe other improvements can be done in the future. (I hidden the previous comment as off-topic)

wxiaoguang commented on Makefile:
```diff
@@ -338,6 +338,8 @@ test: test-frontend test-backend
 test-backend:
 	@echo "Running go test with $(GOTESTFLAGS) -tags '$(TEST_TAGS)'..."
 	@$(GO) test $(GOTESTFLAGS) -tags='$(TEST_TAGS)' $(GO_PACKAGES)
+# Check if the files in ./build are compilable, they could be ignored because of 'go:build ignore'
+	@find ./build -name '*.go' -exec go build -o /dev/null {} \;
```
I am not sure whether it is really needed.

If any one needs the `make fmt` and it breaks in the future, developers can catch the problem in first time.

Otherwise if a tool int `build` is out-dated, I think it can be left there as it is, even if it's broken?

wolfogre replied:
I agree with you, but the `gitea-format-imports.go` isn't a build tool used by Makefile, what should we do with it?

wxiaoguang replied:
The file `gitea-format-imports.go` is quite tricky, it only for debug purpose in the first PR https://github.com/go-gitea/gitea/pull/17684 (somewhat I regret that I committed it into the codebase). 

Although it was designed to do some formatting work, but the work is done by `code-batch-process.go` directly. And still now, I can not find any module requires it in the repository since then.

Maybe we can just remove  `gitea-format-imports.go`  to avoid confuses.

wxiaoguang commented:
Although the change LGTM, I am not sure the `gitea-fmt` is really a good name for it, since it only handles imports now.

ps: there is also `gitea-vet` (https://gitea.com/gitea/gitea-vet) 😁  maybe it's time to improve some toolchains for Gitea together.

Since I have no preference and have no objection, I will put my LGTM here.

wxiaoguang commented on build/codeformat/formatimports.go:
```diff
@@ -197,5 +197,9 @@ func FormatGoImports(file string, doChangedFiles, doWriteFile bool) error {
 	}
 
+	if !doChangedFiles && !doWriteFile {
+		fmt.Printf("%s", formattedBytes)
```
Here seems incorrect, at least, there should be a "\n" at the end. Otherwise the lines from different calls may be outputted in the same line.

And I am not sure do we really need it, it is is necessary, it needs more comments with a real case.

wolfogre replied:
For no particular reason, I just want it works like gofmt, see https://cs.opensource.google/go/go/+/master:src/cmd/gofmt/gofmt.go;l=300

wxiaoguang replied:
I would prefer to keep it as simple as possible if there is no real need.

There was a discussion in https://github.com/go-gitea/gitea/pull/18633#pullrequestreview-874045422 about showing the diff result when doing formatting, till now, there is still no such need.

wolfogre commented:
Or we just remove the broken `gitea-format-imports.go`? It is not used by Makefile, and the command `gitea-fmt` could be replaced by `gitea-vet`.

wxiaoguang commented:
> Or we just remove the broken `gitea-format-imports.go`? It is not used by Makefile, and the command `gitea-fmt` could be replaced by `gitea-vet`.

Agree, removing `gitea-format-imports.go` seems a clear solution. And in the future, `gitea-vet` could take more works (if most developers agree)

wolfogre commented:
Changed the commits, just removed the broken file and the outdated introduction which confused me 😂

wxiaoguang commented:
> Changed the commits, just removed the broken file and the outdated introduction which confused me 😂

Actually, the comment is correct (but not clear ....), I made some changes to it.

wolfogre commented:
> > Changed the commits, just removed the broken file and the outdated introduction which confused me 😂
> 
> Actually, the comment is correct (but not clear ....), I made some changes to it.
> 
> https://github.com/go-gitea/gitea/blob/a2db81063be90499c16cbe1af777ee8858aa192d/build/code-batch-process.go#L267-L273

I see, I dropped the comment deleting.
